code/modules/mob/living/carbon/carbon.dm

```diff
@@ -588,14 +593,20 @@
 	if(!GLOB.tinted_weldhelh)
 		return
 	tinttotal = get_total_tint()
+	clear_fullscreen("tint", 0)
 	if(tinttotal >= TINT_BLIND)
 		become_blind(EYES_COVERED)
-	else if(tinttotal >= TINT_DARKENED)
+	else if(tinttotal >= TINT_DARKENED && tinttotal < TINT_BLIND)
 		cure_blind(EYES_COVERED)
-		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2)
-	else
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2, clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal >= TINT_LIGHTER && tinttotal < TINT_DARKENED)
```

Comment:
awesome stealth sunglasses nerf

Reply:
Just made tint work for these, so it works as it should

Reply:
it's not clearly mentioned in the PR body though

Reply:
Mmm yeah, I wrote that I updated the tint calculations and didn't think about the sunglasses at that time

Reply:
I'll remove the tint from this and maybe I'll do another pr to update it